src/main/java/com/exactpro/th2/act/ActConnections.java

```diff
@@ -22,20 +22,23 @@
 import com.exactpro.th2.common.schema.factory.CommonFactory;
 
 public abstract class ActConnections<C extends CustomConfiguration> {
-
 	protected final RhBatchService handConnector;
 	protected final EventStoreHandler eventStoreHandler;
 
 	protected final CommonFactory commonFactory;
 	protected final C customConfiguration;
 
-	public ActConnections(CommonFactory commonFactory) throws Exception {
+	public ActConnections(CommonFactory commonFactory, RhBatchServiceProvider provider) throws Exception {
 		this.commonFactory = commonFactory;
-		this.handConnector = commonFactory.getGrpcRouter().getService(RhBatchService.class);
+		this.handConnector = provider.provide();
 		this.eventStoreHandler = new EventStoreHandler(commonFactory);
 		this.customConfiguration = createCustomConfiguration(commonFactory);
 	}
 
+	public ActConnections(CommonFactory commonFactory) throws Exception {
```

Comment:
Maybe we should deprecate this constructor if we have another one with RhBatchServiceProvider interface?

Reply:
I think user can choose one of mode to interact with hand until we decide to discard gRPC communication in this case.

Reply:
Okay...